### ISSUE TITLE
Git functionality

### ISSUE THREAD

**Issue by paxnovem** [EddieRingle/hubroid#27]
With android ide's such as touchqode. It would be nice if hubroid could get basic commit functionality.

**Comment by EddieRingle** [comment]
Yes, being able to update files on the go is a feature I've thought about. I'm sure it can be done, but it would take awhile to implement and perfect.

**Comment by EddieRingle** [comment]
Just posting here as a note, I've recommended an API addition that will allow for diffs to be sent in via the API and be applied to a repository. See [here](http://support.github.com/discussions/api/235-send-diffs-to-the-api-to-have-them-committed).

**Comment by EddieRingle** [comment]
This looks to be possible with the addition of the [v3 Git API](http://developer.github.com/v3/git/). :)

**Comment by ProLoser** [comment]
There is another app in the market that adds GIT functionality to your phone and allows you to work on code and push. Perhaps a partnership with that developer (to reduce scope creep) would be beneficial? [agit](https://play.google.com/store/apps/details?id=com.madgag.agit&feature=search_result#?t=W251bGwsMSwyLDEsImNvbS5tYWRnYWcuYWdpdCJd)

**Comment by EddieRingle** [comment]
Actually @ProLoser, Agit is developed by GitHub's own @rtyley. He submitted a patch to the previous version of Hubroid that added cloning support through Agit. I will probably re-implement a similar feature soon, but I've had bigger plans for Hubroid in this area through the new Git APIs GitHub provides.
